sycl/include/CL/sycl/ONEAPI/group_algorithm.hpp

```diff
@@ -190,6 +190,28 @@ using EnableIfIsNonNativeOp = cl::sycl::detail::enable_if_t<
         !cl::sycl::detail::is_native_op<T, BinaryOperation>::value,
     T>;
 
+template <typename T>
+struct is_native_function_object_impl : std::false_type {};
+
+template <template <typename> class BinaryOperation, typename T>
+struct is_native_function_object_impl<BinaryOperation<T>>
+    : std::integral_constant<
+          bool, (sycl::detail::is_scalar_arithmetic<T>::value ||
+                 sycl::detail::is_vector_arithmetic<T>::value ||
+                 std::is_same<T, void>::value) &&
+                    sycl::detail::is_native_op<T, BinaryOperation<T>>::value> {
+};
+
+template <typename T>
+struct is_native_function_object
+    : is_native_function_object_impl<typename std::decay<T>::type> {};
+
+#if __cplusplus >= 201703L
```

Comment:
Template variables are the feature of c++14, that is true. But `inline` variable is the feature of c++17. All traits like ``is_same_v``, ``is_reference_v``, etc. have been added in c++17 and not earlier right because of  ``inline`` variables feature lack. I would not apply this suggestion.

Reply:
You are right.